homeassistant/components/reolink/binary_sensor.py

```diff
@@ -129,8 +129,12 @@ def __init__(
         super().__init__(reolink_data, channel)
         self.entity_description = entity_description
 
-        if self._host.api.model in DUAL_LENS_DUAL_MOTION_MODELS:
-            self._attr_name = f"{entity_description.name} lens {self._channel}"
+        if (
+            self._host.api.model in DUAL_LENS_DUAL_MOTION_MODELS
+            and self._name_translation_key is not None
+        ):
+            name = self.platform.platform_translations.get(self._name_translation_key)
```

Comment:
This is a highly unusual setup. The only way we currently provide for placeholders is devices. 

I don't know the full reolink context, but wouldn't it be better to create a subdevice for every channel? This way the UI also looks a lot clearer (as you don't have n amount of channels multiplied by X amount of sensors in one device, but every channel has it's own list of sensors)

Reply:
Yea this needs some context to be understood properly:
There are 2 senarios:
1) The Reolink integration can be used to connect to a Reolink camera directly: in this case there is only 1 device with all entities.
2) The Reolink integration connects to a Reolink NVR which has multiple (up to 36) camera's connected to it: In this case there is 1 NVR device and 1 device per camera and entities are sorted acordingly.

Things become a bit complicated with the DUAL lens models from Reolink, these are single cameras that house 2 lenses with 2 image sensors.
- The tracking camera's (like TrackMix) are basically a normal single lens camera with only a extra (zoomed in) camera stream, the extra stream (camera entity) is placed under the same device.
- The modern DUO cameras (DUO 2) have 2 lenses but in the firmware the images from both lenses are combined in a single RTSP/FLV/RTMP stream with a ultra wide resolution. From software these are identical to normal cameras
- The old DUO V1 (official sales stopped this year) is the trouble maker. This is the only camera model (of the 32 tested models) that has 2 seperate streams for the two lenses but also seperate motion and AI detection for each lens. Since this is physically 1 device and many features/entities control the behaviour of both lenses, this is represented as a single device. For example the floodlight, record, volume, siren are shared: there is only 1 entity to control the camera recording (which applies to both lenses).

Reply:
So currently this code is only needed for the Reolink DUO V1 model: https://reolink.com/product/reolink-duo-v1/

Reply:
So if I understand correctly, this code would only be used for one model?

Would it then be better to maybe split up that subdevice in another 2 subdevices? That probably also isn't a simple solution, but I think we would avoid creating a custom solution for only 1 integration for only 1 device. (So if something might change in this in the future this might break, instead of the other solution where it's sticking with what we have)

Reply:
Alternatevely I could change the self._attr_translation_key for this model, but that would add 12 extra translations.

Reply:
Yes the code is only used for 1 (depricated) model.

Reply:
Splitting up the device in seperate subdevices is not trivial at all since the other (I think about 15 entities) are shared between the lenses: for instance there is only 1 API endpoint to control the floodlight (which is shared for the 2 lenses).

Reply:
But I mean, one main device for everything, and putting the lens specifics in a separate device

Reply:
In theory yes, but in practice it would result in a lot of extra code since I would need a new base class than can manually seperate out the entities. I kind of do not want to go through the trouble for only this old model.

What about just setting the _attr_name and not translating these entities for this specific (pain in my ***) model?
It is only 1 out of 32 models that is not even sold anymore.

Reply:
I understand, but let's throw this up for discussion as I understand both sides but don't know what's best

Reply:
@joostlek as sugested on discord I implemented one translation key for each.